src/Domain/CoordinateMaps/TimeDependent/ShapeTransitionFunctions.hpp

```diff
@@ -21,32 +28,51 @@ class SphereTransition : public tt::ConformsTo<protocols::TransitionFunc> {
   SphereTransition(double r_min, double r_max)
       : r_min_(r_min),
         r_max_(r_max),
-        a_(-1. / (r_max_ - r_min_)),
-        b_(r_max_ / (r_max_ - r_min)) {}
+        a_(-1. / (r_max - r_min)),
+        b_(r_max / (r_max - r_min)) {
+    if (r_min <= 0.) {
+      ERROR("The minimum radius must be greater than 0 but is " << r_min);
+    }
+    if (r_max <= r_min) {
+      ERROR(
+          "The maximum radius must be greater than the minimum radius but "
+          "r_max =  "
+          << r_max << ", and r_min = " << r_min);
+    }
+  }
 
   explicit SphereTransition() = default;
 
   template <typename T>
-  T operator()(std::array<T, 3> source_coords) const {
-    return a_ + b_ / magnitude(source_coords);
+  T operator()(const std::array<T, 3>& source_coords) const {
+    const T mag = magnitude(source_coords);
+    check_magnitudes(mag);
+    return a_ + b_ / mag;
   }
 
-  double original_radius_over_radius(std::array<double, 3> target_coords,
-                                     double distorted_radius) const {
+  std::optional<double> original_radius_over_radius(
+      const std::array<double, 3>& target_coords,
+      double distorted_radius) const {
     const double mag = magnitude(target_coords);
-    return (mag + distorted_radius * b_) / (1. - distorted_radius * a_) / mag;
+    const double original_radius =
+        (mag + distorted_radius * b_) / (1. - distorted_radius * a_);
+    return original_radius >= r_min_ and original_radius <= r_max_
```

Comment:
What happens if mag==0?
Also, what happens if distorted_radius is 1/a ?
Both of these cases should return std::nullopt I think.